scheduler/src/cook/mesos/api.clj

```diff
@@ -1576,6 +1580,26 @@
       (-> states (set/difference util/instance-states) (conj "completed"))
       states)))
 
+(defn valid-name-filter?
+  "Returns true if the provided name filter is either nil or satisfies the JobNameListFilter schema"
+  [name]
+  (or (nil? name)
+      (nil? (s/check JobNameListFilter name))))
+
+(defn name-filter-str->name-filter-pattern
+  "Generates a regex pattern corresponding to a user-provided name filter string"
+  [name]
+  (re-pattern (-> name
+                  (str/replace #"\." "\\\\.")
+                  (str/replace #"\*+" ".*"))))
+
+(defn name-filter-str->name-filter-fn
+  "Returns a name-filtering function (or nil) given a user-provided name filter string"
+  [name]
+  (if name
+    #(re-matches (name-filter-str->name-filter-pattern name) %)
```

Comment:
Not sure if this actually matters but would it make a difference to compile the regex once and re-use it?

Reply:
Good idea, done